src/main/java/kitchenpos/application/OrderService.java

```diff
@@ -1,108 +1,121 @@
         final List<OrderLineItem> savedOrderLineItems = new ArrayList<>();
-        for (final OrderLineItem orderLineItem : orderLineItems) {
-            orderLineItem.setOrderId(orderId);
-            savedOrderLineItems.add(orderLineItemDao.save(orderLineItem));
+        for (final OrderLineItemRequest orderLineItemRequest : orderLineItemsRequests) {
+            Menu menu = menuRepository.findById(orderLineItemRequest.getMenuId())
+                .orElseThrow(IllegalArgumentException::new);
+            OrderLineItem orderLineItem = new OrderLineItem(orderId, menu,
+                orderLineItemRequest.getQuantity());
+            savedOrderLineItems.add(orderLineItemRepository.save(orderLineItem));
         }
-        savedOrder.setOrderLineItems(savedOrderLineItems);
+        savedOrder.updateOrderLineItems(savedOrderLineItems);
 
-        return savedOrder;
+        return getOrderResponse(savedOrder);
     }
 
-    public List<Order> list() {
-        final List<Order> orders = orderDao.findAll();
+
+    public List<OrderResponse> list() {
+        final List<Order> orders = orderRepository.findAll();
 
         for (final Order order : orders) {
-            order.setOrderLineItems(orderLineItemDao.findAllByOrderId(order.getId()));
+            order.updateOrderLineItems(orderLineItemRepository.findAllByOrderId(order.getId()));
         }
 
-        return orders;
+        return orders.stream().map(this::getOrderResponse).collect(Collectors.toList());
     }
 
     @Transactional
-    public Order changeOrderStatus(final Long orderId, final Order order) {
-        final Order savedOrder = orderDao.findById(orderId)
-                .orElseThrow(IllegalArgumentException::new);
+    public OrderResponse changeOrderStatus(final Long orderId, final OrderRequest order) {
+        final Order savedOrder = orderRepository.findById(orderId)
+            .orElseThrow(IllegalArgumentException::new);
 
-        if (Objects.equals(OrderStatus.COMPLETION.name(), savedOrder.getOrderStatus())) {
+        if (Objects.equals(OrderStatus.COMPLETION, savedOrder.getOrderStatus())) {
             throw new IllegalArgumentException();
         }
 
         final OrderStatus orderStatus = OrderStatus.valueOf(order.getOrderStatus());
-        savedOrder.setOrderStatus(orderStatus.name());
+        savedOrder.changeOrderStstus(orderStatus);
 
-        orderDao.save(savedOrder);
+        orderRepository.save(savedOrder);
 
-        savedOrder.setOrderLineItems(orderLineItemDao.findAllByOrderId(orderId));
+        savedOrder.updateOrderLineItems(orderLineItemRepository.findAllByOrderId(orderId));
+
+        return getOrderResponse(savedOrder);
+    }
 
-        return savedOrder;
+    private OrderResponse getOrderResponse(Order savedOrder) {
+        List<OrderLineItemResponse> orderLineItemResponses = savedOrder.getOrderLineItems().stream()
+            .map(OrderLineItemResponse::of).collect(Collectors.toList());
```

Comment:
람다와 같이 함수가 체이닝되는 코드는 한줄에 붙여 쓰는 것 보다 메서드마다 하나의 라인으로 줄바꿈해서 표시하면 더 가독성 좋은 코드가 됩니다.

```suggestion
        List<OrderLineItemResponse> orderLineItemResponses = savedOrder.getOrderLineItems()
            .stream()
            .map(OrderLineItemResponse::of)
            .collect(Collectors.toList());
```